src/config.rs

```diff
@@ -33,11 +36,11 @@ pub struct ConfigBuilder {
     use_tls: bool,
     async_buffer_size: Option<usize>,
     buffer_size: Option<usize>,
-    buffer_duration: Option<u64>,
+    buffer_duration: Option<Duration>,
     additional_fields: BTreeMap<Value, Value>,
     full_buffer_policy: Option<FullBufferPolicy>,
-    connect_timeout_ms: Option<u64>,
-    write_timeout_ms: Option<u64>,
+    connect_timeout_ms: Option<Duration>,
```

Comment:
```suggestion
    connect_timeout: Option<Duration>,
```
I don't think the unit suffix is required anymore